packages/react-native-platform-override/src/BatchingQueue.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+/**
+ * Executes actions, attempting to group by a given key
+ */
+export default class BatchingQueue<TKey> {
+  private keyedQueues: Map<TKey, Array<() => Promise<void>>> = new Map();
+  private currentKey?: TKey;
+
+  enqueue<T>(key: TKey, action: () => Promise<T>): Promise<T> {
+    return new Promise((resolve, reject) => {
+      if (!this.keyedQueues.has(key)) {
+        this.keyedQueues.set(key, []);
+      }
+
+      this.keyedQueues.get(key).push(async () => {
+        try {
+          resolve(await action());
+        } catch (ex) {
+          reject(ex);
+        }
+      });
+
+      if (!this.currentKey) {
+        this.currentKey = key;
+        this.pumpQueue();
+      }
+    });
+  }
+
+  private async pumpQueue(): Promise<void> {
+    const currentQueue = this.keyedQueues.get(this.currentKey!);
+
+    while (currentQueue.length > 0) {
+      await currentQueue.shift()!();
+    }
+
+    this.keyedQueues.delete(this.currentKey!);
+    this.currentKey = undefined;
+
+    if (this.keyedQueues.size > 0) {
```

Comment:
>this [](start = 8, length = 4)

nit: Perhaps add a comment that says something like: finding they next key with the largest queue to process next.

Reply:
Can add this